Log s7comm device information

mvhensbergen opened the pull request:
## 🗣 Description ##

These changes create a new log file 's7comm_known_devices.log' which contains s7comm device identification information obtained from the s7comm protocol.

## 💭 Motivation and context ##

In relation to OT Network Detection and Response, device and asset enumeration obtained by passive network inspection can be a good tool to monitor network devices. Baselining on identification strings allow alerting on unintentional changes in e.g. firmware.

This code adds additional binpac structures to more obtain the information in question.

## 🧪 Testing ##

A test trace has been added to the repository and I have done field testing in a few OT environments.

An existing testcase fails after my change but I believe the new parsing is more accurate than the original one. According to my analysis, the original parser incorrectly parsed a packet leading to an extra log line.

Feel free to contact me to discuss.

mvhensbergen commented:
Is there any update on the review of this PR?

mmguero commented:
When I install and it runs the btest, I am getting some failures:

```bash
.root 18bec1a4632a tests> btest 
[ 50%] analyzer.s7ident ... failed
[ 75%] analyzer.snap7 ... failed
2 of 4 tests failed
```

```bash
.root 18bec1a4632a tests> btest 
[ 50%] analyzer.s7ident ... failed
[ 75%] analyzer.snap7 ... failed
2 of 4 tests failed
.root 18bec1a4632a tests> btest -A
[  0%] analyzer.availability ... ok
[ 25%] analyzer.s7comm_plus ... ok
[ 50%] analyzer.s7ident ... failed
[ 75%] analyzer.snap7 ... failed
2 of 4 tests failed
.root 18bec1a4632a tests> btest -A -d
[  0%] analyzer.availability ... ok
[ 25%] analyzer.s7comm_plus ... ok
[ 50%] analyzer.s7ident ... failed
  % 'btest-diff s7comm_plus.log' failed unexpectedly (exit code 1)
  % cat .diag
  btest-diff: input s7comm_plus.log does not exist.

  % cat .stderr

[ 75%] analyzer.snap7 ... failed
  % 'btest-diff s7comm.log' failed unexpectedly (exit code 1)
  % cat .diag
  == File ===============================
  #separator \x09
  #set_separator	,
  #empty_field	(empty)
  #unset_field	-
  #path	s7comm
  #open	2025-02-27-14-29-57
  #fields	ts	uid	id.orig_h	id.orig_p	id.resp_h	id.resp_p	is_orig	source_h	source_p	destination_h	destination_p	rosctr_code	rosctr_name	pdu_reference	function_code	function_name	subfunction_code	subfunction_name	error_class	error_code
  #types	time	string	addr	port	addr	port	bool	addr	port	addr	port	count	string	count	string	string	string	string	string	string
  1454969289.930413	CHhAvVGS1DHFjwGM9	134.217.61.131	51212	134.217.61.211	102	T	134.217.61.131	51212	134.217.61.211	102	1	Job-Request	0	0xf0	Setup Communication	-	-	-	-
  1454969289.933284	CHhAvVGS1DHFjwGM9	134.217.61.131	51212	134.217.61.211	102	F	134.217.61.211	102	134.217.61.131	51212	3	ACK-Data	0	0xf0	Setup Communication	-	-	No error	0x00
  1454969289.937630	CHhAvVGS1DHFjwGM9	134.217.61.131	51212	134.217.61.211	102	T	134.217.61.131	51212	134.217.61.211	102	7	User-Data	1	0x44	Request: CPU Functions	0x01	Read SZL	-	-
  1454969289.942358	CHhAvVGS1DHFjwGM9	134.217.61.131	51212	134.217.61.211	102	F	134.217.61.211	102	134.217.61.131	51212	7	User-Data	1	0x84	Response: CPU Functions	0x01	Read SZL	No error	0x00
  1454969289.946656	CHhAvVGS1DHFjwGM9	134.217.61.131	51212	134.217.61.211	102	T	134.217.61.131	51212	134.217.61.211	102	7	User-Data	2	0x44	Request: CPU Functions	0x01	Read SZL	-	-
  1454969289.955381	CHhAvVGS1DHFjwGM9	134.217.61.131	51212	134.217.61.211	102	F	134.217.61.211	102	134.217.61.131	51212	7	User-Data	3	0x84	Response: CPU Functions	0x01	Read SZL	No error	0x00
  1454969289.960281	CHhAvVGS1DHFjwGM9	134.217.61.131	51212	134.217.61.211	102	T	134.217.61.131	51212	134.217.61.211	102	7	User-Data	4	0x44	Request: CPU Functions	0x01	Read SZL	-	-
  1454969289.965341	CHhAvVGS1DHFjwGM9	134.217.61.131	51212	134.217.61.211	102	F	134.217.61.211	102	134.217.61.131	51212	7	User-Data	4	0x84	Response: CPU Functions	0x01	Read SZL	No error	0x00
  1454969289.967206	CHhAvVGS1DHFjwGM9	134.217.61.131	51212	134.217.61.211	102	T	134.217.61.131	51212	134.217.61.211	102	7	User-Data	5	0x44	Request: CPU Functions	0x01	Read SZL	-	-
  1454969289.972343	CHhAvVGS1DHFjwGM9	134.217.61.131	51212	134.217.61.211	102	F	134.217.61.211	102	134.217.61.131	51212	7	User-Data	5	0x84	Response: CPU Functions	0x01	Read SZL	No error	0x00
  1454969289.974084	CHhAvVGS1DHFjwGM9	134.217.61.131	51212	134.217.61.211	102	T	134.217.61.131	51212	134.217.61.211	102	7	User-Data	6	0x44	Request: CPU Functions	0x01	Read SZL	-	-
  1454969289.979368	CHhAvVGS1DHFjwGM9	134.217.61.131	51212	134.217.61.211	102	F	134.217.61.211	102	134.217.61.131	51212	7	User-Data	6	0x84	Response: CPU Functions	0x01	Read SZL	No error	0x00
  1454969289.994223	CHhAvVGS1DHFjwGM9	134.217.61.131	51212	134.217.61.211	102	T	134.217.61.131	51212	134.217.61.211	102	7	User-Data	7	0x43	Request: Block Functions	0x01	List Blocks	-	-
  1454969290.002344	CHhAvVGS1DHFjwGM9	134.217.61.131	51212	134.217.61.211	102	F	134.217.61.211	102	134.217.61.131	51212	7	User-Data	7	0x83	Response: Block Functions	0x01	List Blocks	No error	0x00
  1454969290.008331	CHhAvVGS1DHFjwGM9	134.217.61.131	51212	134.217.61.211	102	T	134.217.61.131	51212	134.217.61.211	102	1	Job-Request	8	0x1d	Start Upload	-	-	-	-
  1454969290.013372	CHhAvVGS1DHFjwGM9	134.217.61.131	51212	134.217.61.211	102	F	134.217.61.211	102	134.217.61.131	51212	3	ACK-Data	8	0x1d	Start Upload	-	-	No error	0x00
  1454969290.014330	CHhAvVGS1DHFjwGM9	134.217.61.131	51212	134.217.61.211	102	T	134.217.61.131	51212	134.217.61.211	102	1	Job-Request	9	0x1e	Upload	-	-	-	-
  1454969290.017399	CHhAvVGS1DHFjwGM9	134.217.61.131	51212	134.217.61.211	102	F	134.217.61.211	102	134.217.61.131	51212	3	ACK-Data	9	0x1e	Upload	-	-	No error	0x00
  1454969290.018338	CHhAvVGS1DHFjwGM9	134.217.61.131	51212	134.217.61.211	102	T	134.217.61.131	51212	134.217.61.211	102	1	Job-Request	10	0x1f	End Upload	-	-	-	-
  1454969290.021307	CHhAvVGS1DHFjwGM9	134.217.61.131	51212	134.217.61.211	102	F	134.217.61.211	102	134.217.61.131	51212	3	ACK-Data	10	0x1f	End Upload	-	-	No error	0x00
  1454969290.039663	CHhAvVGS1DHFjwGM9	134.217.61.131	51212	134.217.61.211	102	T	134.217.61.131	51212	134.217.61.211	102	7	User-Data	11	0x43	Request: Block Functions	0x03	Get Block Info	-	-
  1454969290.043334	CHhAvVGS1DHFjwGM9	134.217.61.131	51212	134.217.61.211	102	F	134.217.61.211	102	134.217.61.131	51212	7	User-Data	11	0x83	Response: Block Functions	0x03	Get Block Info	Parameter Error	0xd209
  1454969290.045028	CHhAvVGS1DHFjwGM9	134.217.61.131	51212	134.217.61.211	102	T	134.217.61.131	51212	134.217.61.211	102	1	Job-Request	12	0x1d	Start Upload	-	-	-	-
  1454969290.049325	CHhAvVGS1DHFjwGM9	134.217.61.131	51212	134.217.61.211	102	F	134.217.61.211	102	134.217.61.131	51212	3	ACK-Data	12	0x1d	Start Upload	-	-	No error	0x00
  1454969290.050318	CHhAvVGS1DHFjwGM9	134.217.61.131	51212	134.217.61.211	102	T	134.217.61.131	51212	134.217.61.211	102	1	Job-Request	13	0x1e	Upload	-	-	-	-
  1454969290.053388	CHhAvVGS1DHFjwGM9	134.217.61.131	51212	134.217.61.211	102	F	134.217.61.211	102	134.217.61.131	51212	3	ACK-Data	13	0x1e	Upload	-	-	No error	0x00
  1454969290.054316	CHhAvVGS1DHFjwGM9	134.217.61.131	51212	134.217.61.211	102	T	134.217.61.131	51212	134.217.61.211	102	1	Job-Request	14	0x1f	End Upload	-	-	-	-
  1454969290.057336	CHhAvVGS1DHFjwGM9	134.217.61.131	51212	134.217.61.211	102	F	134.217.61.211	102	134.217.61.131	51212	3	ACK-Data	14	0x1f	End Upload	-	-	No error	0x00
  1454969290.162497	CHhAvVGS1DHFjwGM9	134.217.61.131	51212	134.217.61.211	102	T	134.217.61.131	51212	134.217.61.211	102	1	Job-Request	15	0x1d	Start Upload	-	-	-	-
  1454969290.166337	CHhAvVGS1DHFjwGM9	134.217.61.131	51212	134.217.61.211	102	F	134.217.61.211	102	134.217.61.131	51212	3	ACK-Data	15	0x1d	Start Upload	-	-	No error	0x00
  1454969290.166367	CHhAvVGS1DHFjwGM9	134.217.61.131	51212	134.217.61.211	102	T	134.217.61.131	51212	134.217.61.211	102	1	Job-Request	16	0x1e	Upload	-	-	-	-
  1454969290.171260	CHhAvVGS1DHFjwGM9	134.217.61.131	51212	134.217.61.211	102	F	134.217.61.211	102	134.217.61.131	51212	3	ACK-Data	16	0x1e	Upload	-	-	No error	0x00
  1454969290.171327	CHhAvVGS1DHFjwGM9	134.217.61.131	51212	134.217.61.211	102	T	134.217.61.131	51212	134.217.61.211	102	1	Job-Request	17	0x1f	End Upload	-	-	-	-
  1454969290.174326	CHhAvVGS1DHFjwGM9	134.217.61.131	51212	134.217.61.211	102	F	134.217.61.211	102	134.217.61.131	51212	3	ACK-Data	17	0x1f	End Upload	-	-	No error	0x00
  1454969290.192363	CHhAvVGS1DHFjwGM9	134.217.61.131	51212	134.217.61.211	102	T	134.217.61.131	51212	134.217.61.211	102	1	Job-Request	18	0x1d	Start Upload	-	-	-	-
  1454969290.196331	CHhAvVGS1DHFjwGM9	134.217.61.131	51212	134.217.61.211	102	F	134.217.61.211	102	134.217.61.131	51212	3	ACK-Data	18	0x1d	Start Upload	-	-	No error	0x00
  1454969290.196356	CHhAvVGS1DHFjwGM9	134.217.61.131	51212	134.217.61.211	102	T	134.217.61.131	51212	134.217.61.211	102	1	Job-Request	19	0x1e	Upload	-	-	-	-
  1454969290.200386	CHhAvVGS1DHFjwGM9	134.217.61.131	51212	134.217.61.211	102	F	134.217.61.211	102	134.217.61.131	51212	3	ACK-Data	19	0x1e	Upload	-	-	No error	0x00
  1454969290.201327	CHhAvVGS1DHFjwGM9	134.217.61.131	51212	134.217.61.211	102	T	134.217.61.131	51212	134.217.61.211	102	1	Job-Request	20	0x1f	End Upload	-	-	-	-
  1454969290.204294	CHhAvVGS1DHFjwGM9	134.217.61.131	51212	134.217.61.211	102	F	134.217.61.211	102	134.217.61.131	51212	3	ACK-Data	20	0x1f	End Upload	-	-	No error	0x00
  1454969290.309923	CHhAvVGS1DHFjwGM9	134.217.61.131	51212	134.217.61.211	102	T	134.217.61.131	51212	134.217.61.211	102	1	Job-Request	21	0x1d	Start Upload	-	-	-	-
  1454969290.314054	CHhAvVGS1DHFjwGM9	134.217.61.131	51212	134.217.61.211	102	T	134.217.61.131	51212	134.217.61.211	102	7	User-Data	22	0x47	Request: Time Functions	0x01	Read Clock	-	-
  1454969290.317335	CHhAvVGS1DHFjwGM9	134.217.61.131	51212	134.217.61.211	102	F	134.217.61.211	102	134.217.61.131	51212	7	User-Data	22	0x87	Response: Time Functions	0x01	Read Clock	No error	0x00
  1454969290.319495	CHhAvVGS1DHFjwGM9	134.217.61.131	51212	134.217.61.211	102	T	134.217.61.131	51212	134.217.61.211	102	7	User-Data	23	0x47	Request: Time Functions	0x02	Set Clock	-	-
  1454969290.322351	CHhAvVGS1DHFjwGM9	134.217.61.131	51212	134.217.61.211	102	F	134.217.61.211	102	134.217.61.131	51212	7	User-Data	23	0x87	Response: Time Functions	0x02	Set Clock	No error	0x00
  1454969290.324047	CHhAvVGS1DHFjwGM9	134.217.61.131	51212	134.217.61.211	102	T	134.217.61.131	51212	134.217.61.211	102	1	Job-Request	24	0x05	Write Variable	-	-	-	-
  1454969290.327352	CHhAvVGS1DHFjwGM9	134.217.61.131	51212	134.217.61.211	102	F	134.217.61.211	102	134.217.61.131	51212	3	ACK-Data	24	0x05	Write Variable	-	-	No error	0x00
  1454969290.328136	CHhAvVGS1DHFjwGM9	134.217.61.131	51212	134.217.61.211	102	T	134.217.61.131	51212	134.217.61.211	102	1	Job-Request	25	0x04	Read Variable	-	-	-	-
  1454969290.331336	CHhAvVGS1DHFjwGM9	134.217.61.131	51212	134.217.61.211	102	F	134.217.61.211	102	134.217.61.131	51212	3	ACK-Data	25	0x04	Read Variable	-	-	No error	0x00
  1454969290.332164	CHhAvVGS1DHFjwGM9	134.217.61.131	51212	134.217.61.211	102	T	134.217.61.131	51212	134.217.61.211	102	1	Job-Request	26	0x05	Write Variable	-	-	-	-
  1454969290.335345	CHhAvVGS1DHFjwGM9	134.217.61.131	51212	134.217.61.211	102	F	134.217.61.211	102	134.217.61.131	51212	3	ACK-Data	26	0x05	Write Variable	-	-	No error	0x00
  1454969290.336133	CHhAvVGS1DHFjwGM9	134.217.61.131	51212	134.217.61.211	102	T	134.217.61.131	51212	134.217.61.211	102	1	Job-Request	27	0x04	Read Variable	-	-	-	-
  1454969290.339352	CHhAvVGS1DHFjwGM9	134.217.61.131	51212	134.217.61.211	102	F	134.217.61.211	102	134.217.61.131	51212	3	ACK-Data	27	0x04	Read Variable	-	-	No error	0x00
  1454969290.350892	CHhAvVGS1DHFjwGM9	134.217.61.131	51212	134.217.61.211	102	T	134.217.61.131	51212	134.217.61.211	102	1	Job-Request	28	0x29	PLC Stop	-	-	-	-
  1454969290.356402	CHhAvVGS1DHFjwGM9	134.217.61.131	51212	134.217.61.211	102	F	134.217.61.211	102	134.217.61.131	51212	3	ACK-Data	28	0x29	PLC Stop	-	-	No error	0x00
  1454969290.358155	CHhAvVGS1DHFjwGM9	134.217.61.131	51212	134.217.61.211	102	T	134.217.61.131	51212	134.217.61.211	102	1	Job-Request	29	0x28	PLC Control	_MODU	PLC Copy Ram to Rom	-	-
  1454969290.458027	CHhAvVGS1DHFjwGM9	134.217.61.131	51212	134.217.61.211	102	F	134.217.61.211	102	134.217.61.131	51212	3	ACK-Data	29	0x28	PLC Control	-	-	No error	0x00
  1454969290.459216	CHhAvVGS1DHFjwGM9	134.217.61.131	51212	134.217.61.211	102	T	134.217.61.131	51212	134.217.61.211	102	1	Job-Request	30	0x28	PLC Control	_GARB	Compress PLC memory	-	-
  1454969290.466059	CHhAvVGS1DHFjwGM9	134.217.61.131	51212	134.217.61.211	102	F	134.217.61.211	102	134.217.61.131	51212	3	ACK-Data	30	0x28	PLC Control	-	-	No error	0x00
  1454969290.467071	CHhAvVGS1DHFjwGM9	134.217.61.131	51212	134.217.61.211	102	T	134.217.61.131	51212	134.217.61.211	102	1	Job-Request	31	0x28	PLC Control	P_PROGRAM	PLC Start / Stop	-	-
  1454969290.479087	CHhAvVGS1DHFjwGM9	134.217.61.131	51212	134.217.61.211	102	F	134.217.61.211	102	134.217.61.131	51212	3	ACK-Data	31	0x28	PLC Control	-	-	No error	0x00
  #close	2025-02-27-14-29-57
  == Diff ===============================
  --- /dev/fd/63	2025-02-27 14:29:57.113561567 +0000
  +++ /dev/fd/62	2025-02-27 14:29:57.113561567 +0000
  @@ -11,7 +11,6 @@
   XXXXXXXXXX.XXXXXX	CHhAvVGS1DHFjwGM9	134.217.61.131	51212	134.217.61.211	102	T	134.217.61.131	51212	134.217.61.211	102	7	User-Data	1	0x44	Request: CPU Functions	0x01	Read SZL	-	-
   XXXXXXXXXX.XXXXXX	CHhAvVGS1DHFjwGM9	134.217.61.131	51212	134.217.61.211	102	F	134.217.61.211	102	134.217.61.131	51212	7	User-Data	1	0x84	Response: CPU Functions	0x01	Read SZL	No error	0x00
   XXXXXXXXXX.XXXXXX	CHhAvVGS1DHFjwGM9	134.217.61.131	51212	134.217.61.211	102	T	134.217.61.131	51212	134.217.61.211	102	7	User-Data	2	0x44	Request: CPU Functions	0x01	Read SZL	-	-
  -XXXXXXXXXX.XXXXXX	CHhAvVGS1DHFjwGM9	134.217.61.131	51212	134.217.61.211	102	F	134.217.61.211	102	134.217.61.131	51212	7	User-Data	2	0x84	Response: CPU Functions	0x01	Read SZL	No error	0x00
   XXXXXXXXXX.XXXXXX	CHhAvVGS1DHFjwGM9	134.217.61.131	51212	134.217.61.211	102	F	134.217.61.211	102	134.217.61.131	51212	7	User-Data	3	0x84	Response: CPU Functions	0x01	Read SZL	No error	0x00
   XXXXXXXXXX.XXXXXX	CHhAvVGS1DHFjwGM9	134.217.61.131	51212	134.217.61.211	102	T	134.217.61.131	51212	134.217.61.211	102	7	User-Data	4	0x44	Request: CPU Functions	0x01	Read SZL	-	-
   XXXXXXXXXX.XXXXXX	CHhAvVGS1DHFjwGM9	134.217.61.131	51212	134.217.61.211	102	F	134.217.61.211	102	134.217.61.131	51212	7	User-Data	4	0x84	Response: CPU Functions	0x01	Read SZL	No error	0x00
  =======================================

  % cat .stderr

2 of 4 tests failed
```

```
$ .root 18bec1a4632a logs> zeek -C -r /opt/zeek/var/lib/zkg/clones/package/icsnpp-s7comm/tests/traces/s7ident.pcap local
.root 18bec1a4632a logs> ls -ltra
total 114688
drwxrwxrwt 1 root root    39 Feb 27 14:30 ..
drwxr-xr-x 2 root root  4096 Feb 27 14:30 .
-rw-r--r-- 1 root root 22053 Feb 27 14:30 telemetry.log
-rw-r--r-- 1 root root   810 Feb 27 14:30 stats.log
-rw-r--r-- 1 root root  1237 Feb 27 14:30 s7comm_upload_download.log
-rw-r--r-- 1 root root  2061 Feb 27 14:30 s7comm_read_szl.log
-rw-r--r-- 1 root root   542 Feb 27 14:30 s7comm_known_devices.log
-rw-r--r-- 1 root root  3351 Feb 27 14:30 s7comm.log
-rw-r--r-- 1 root root   278 Feb 27 14:30 packet_filter.log
-rw-r--r-- 1 root root 33565 Feb 27 14:30 loaded_scripts.log
-rw-r--r-- 1 root root   275 Feb 27 14:30 known_services.log
-rw-r--r-- 1 root root   202 Feb 27 14:30 known_hosts.log
-rw-r--r-- 1 root root  3126 Feb 27 14:30 cotp.log
-rw-r--r-- 1 root root   642 Feb 27 14:30 conn.log
-rw-r--r-- 1 root root   277 Feb 27 14:30 capture_loss.log
-rw-r--r-- 1 root root  1084 Feb 27 14:30 analyzer.log
```

So it looks like in the test `analyzer.s7ident` the issue is the test results are expecting [`s7comm_plus.log`](https://github.com/mvhensbergen/icsnpp-s7comm/blob/fe5f7636a58806939ff288ae1961728e9a3aa305/tests/analyzer/s7ident.zeek#L4) to be generated, which it is not, so the test fails. However, I ran that PCAP against the `main` branch of the plugin as well, and it wasn't generated there either, so I'm not sure why the test is expecting that in the first place. It could be that you could just remove that line from `s7ident.zeek` and just have the test looking for the remainder of the logs, including your knew logfile, and that could be okay.

From what I can see from the now-failing snap7 test, the `s7comm.log` file generated by your branch contains one fewer lines than it does with the current `main`:

```diff
6c6
< #open	2025-02-27-14-41-08
---
> #open	2025-02-27-14-39-55
14d13
< ##########.######	C#################	134.217.61.131	51212	134.217.61.211	102	F	134.217.61.211	102	134.217.61.131	51212	7	User-Data	2	0x84	Response: CPU Functions	0x01	Read SZL	No error	0x00
71c70
< #close	2025-02-27-14-41-08
---
> #close	2025-02-27-14-39-55
```

Any idea why this would be?

mmguero commented on tests/analyzer/s7ident.zeek:
```diff
@@ -0,0 +1,9 @@
+# @TEST-EXEC: zeek -C -r ${TRACES}/s7ident.pcap %INPUT
+# @TEST-EXEC: btest-diff cotp.log
+# @TEST-EXEC: btest-diff s7comm.log
+# @TEST-EXEC: btest-diff s7comm_plus.log
```
This `s7comm_plus.log` file is not generated when running your branch against `s7ident.pcap`, at least not under Zeek 7.1.0, so this test fails. But even the current `main` branch doesn't generate that log file either, so I'm not sure why you're expecting it here?

mmguero commented on tests/baseline/analyzer.s7ident/s7comm_plus.log:
This file is not generated when the test is run for me when I install your branch when the plugin is installed in the latest stable zeek (v7.1.0). What version are your running, do you know why it would be generated for you and when I test it?

mmguero reviewed (changes requested):
see [my comments](https://github.com/cisagov/icsnpp-s7comm/pull/18#issuecomment-2688169818) on the PR

mvhensbergen commented:
Thank you very much for your feedback! I have taken the time to look at it and here's my analysis.

Firstly, you were complete correct that the s7comm_plus.log file should not have been included in the s7ident test so I removed that.

Regarding the existing testcase that failed, this is my analysis. I made the testcases work again by removing two lines from two log files. While this may seem as a reduction of functionality, I tend to believe that the removal of these lines is correct.

When looking at Wireshark I see that the loglines were generated on a *fragment* of a packet and not the completely reassembled packet. Since the PDU definition used to end with 'bytestring of the rest of the data' the packet would get parsed even though it wasn't complete. Now that I added a more stringent parsing on the identification string, the parsing fails on that packet because it doesnt adhere completely to the slz record spec.

So I think that technically the parser is more correct by not generating a log line for an incomplete s7comm PDU. The issue of handling reassembled frames correctly is a separate issue which is not in the scope of this change.

What are your thoughts on this?

mmguero commented:
We're good with it now!
